Vector.c

```diff
@@ -200,15 +192,13 @@ void Vector_insertionSort(Vector* this) {
    assert(Vector_isConsistent(this));
 }
 
-static void Vector_checkArraySize(Vector* this) {
+static void Vector_resizeIfNecessary(Vector* this, int newSize) {
+   assert(newSize >= 0);
    assert(Vector_isConsistent(this));
-   if (this->items >= this->arraySize) {
-      //int i;
-      //i = this->arraySize;
-      this->arraySize = this->items + this->growthRate;
-      this->array = (Object**) xRealloc(this->array, sizeof(Object*) * this->arraySize);
-      //for (; i < this->arraySize; i++)
-      //   this->array[i] = NULL;
+   if (newSize > this->arraySize) {
+      int oldSize = this->arraySize;
+      this->arraySize = newSize + this->growthRate;
+      this->array =(Object **) xReallocArrayZero(this->array, oldSize, this->arraySize, sizeof(Object*));
```

Comment:
```suggestion
      this->array = (Object **)xReallocArrayZero(this->array, oldSize, this->arraySize, sizeof(Object*));
```
Minor code style note …

Reply:
is there any way to check the formatting? 
The astyle command from the style guide results in another different format:
`this->array = (Object**)xReallocArrayZero(this->array, oldSize, this->arraySize, sizeof(Object*));`